File: solutions/c/resistor-color/2/resistor_color.c
#include "resistor_color.h"

uint16_t color_code(resistor_band_t color)
{
    return color;
}

__attribute__((const)) resistor_band_t* colors(void)
{
    const int length = WHITE + 1;
    static resistor_band_t color_values[length];

    if (color_values[WHITE] == WHITE)
        return color_values;

    for (int i = 0; i < length; i++)
        color_values[i] = (resistor_band_t)i;

    return color_values;
}

File: solutions/csharp/resistor-color/2/ResistorColor.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ResistorColor
{
    private static readonly StringComparison Comparison
        = StringComparison.OrdinalIgnoreCase;

    private static IEnumerable<string> _resistors
        => new[] { "black", "brown", "red", "orange", "yellow",
             "green", "blue", "violet", "grey", "white" };

    public static int ColorCode(string color)
        => _resistors
            .ToList()
            .FindIndex(r => r.Equals(color, Comparison));

    public static string[] Colors()
        => _resistors.ToArray();
}